src/EnergyPlus/DaylightingDevices.cc

```diff
@@ -1620,6 +1614,164 @@ namespace DaylightingDevices {
+                ((zShelfMin < zWinMin) && (zShelfMax > zWinMax))) {   // Shelf goes both above AND below the window
+                zShelfAvg = 0.5 * (zShelfMin + zShelfMax);
+            } else if (zShelfMin < zWinMin) { // Shelf goes partially below the window only
+                Real64 fracAbove = 0.0;
+                if (zShelfMax > zShelfMin) {
+                    fracAbove = (zShelfMax - zWinMin) / (zShelfMax - zShelfMin);
+                    if (fracAbove > 1.0) fracAbove = 1.0;
+                }
+                zShelfAvg = zWinMin + fracAbove * (zShelfMax - zWinMin);
+            } else { // (zShelfMax > zWinMax): Shelf goes partially above window
+                Real64 fracBelow = 0.0;
+                if (zShelfMax > zShelfMin) {
+                    fracBelow = (zWinMax - zShelfMin) / (zShelfMax - zShelfMin);
+                }
+                zShelfAvg = zWinMax - fracBelow * (zWinMax - zShelfMin);
+            }
+
+            // Find height ratio based on shelf average height
+            Real64 heightRatio;
+            if (zWinMax > zWinMin) { // Window has a positive height
+                heightRatio = (zShelfAvg - zWinMin) / (zWinMax - zWinMin);
+                heightRatio = min(heightRatio, 1.0);
+                heightRatio = max(heightRatio, 0.0);
+            } else { // Window does not have a positive height (not realistic) so set height ratio based on shelf location
+                if (zShelfAvg > zWinMax) {
+                    heightRatio = 1.0;
+                } else {
+                    heightRatio = 0.0;
+                }
+            }
+
+            // Take what is left over after the view to shelf is subtracted and then distribute/adjust that proportionally
+            // for the views to ground and sky based on their original weights.  Finally, account for the location of the shelf
+            // with respect to the shelf and reset the values of the actual variables used in the rest of the simulation.
+            leftoverViewFactor = 1.0 - viewFactorToShelf; // By previous logic above, leftover is greater than zero and less than one
+            Real64 vfGroundAdjustMax;
+            Real64 vfGroundAdjustMin;
+            if (viewFactorToGround > viewFactorToShelf) { // How much view to ground could be reduced potentially if shelf at bottom
+                vfGroundAdjustMin = viewFactorToGround - viewFactorToShelf;
+            } else {
+                vfGroundAdjustMin = 0.0;
+            }
+            if (viewFactorToGround > leftoverViewFactor) { // How much view to ground could be reduced potentially if shelf at top
+                vfGroundAdjustMax = leftoverViewFactor;
+            } else {
+                vfGroundAdjustMax = viewFactorToGround;
+            }
+            viewFactorToGround = vfGroundAdjustMin + heightRatio * (vfGroundAdjustMax - vfGroundAdjustMin);
+            viewFactorToSky = leftoverViewFactor - viewFactorToGround;
+        }
+    }
```

Comment:
After all done for this case, maybe a little bit extra (optional) warning that the estimate may not be a precise calculation; (and also there could be scenarios where the view from the window to the building itself which might complicate the case as well)?

Reply:
See previous comment.  I'll put one additional warning at the end of the routine since if it gets there it's because there was something reduced and the user really needs to think through things a bit more.